lib/utils/cert/selfsigned.go

```diff
@@ -50,7 +50,7 @@ type Credentials struct {
 
 // GenerateSelfSignedCert generates a self-signed certificate that
 // is valid for given domain names and ips, returns PEM-encoded bytes with key and cert
-func GenerateSelfSignedCert(hostNames []string) (*Credentials, error) {
+func GenerateSelfSignedCert(hostNames []string, ipAddresses ...string) (*Credentials, error) {
```

Comment:
I wonder if using a variadic parameter here for the `ipAddresses` will make it harder to add new parameters in future. I'd probably just go for `[]string`.

Reply:
This is used in a bunch of places so we'd have to change those calls I believe. Happy to do that. Just was trying to minimize impact.

Reply:
I think whilst that minimises impact now - it's only creating an impact in future if we need to add additional parameters. As far as I can tell, there's only two actual calls to this function, and only three in tests. They'd only need updating to include an empty slice for ipAddresses.

Reply:
I've updated the calls.  Please take a look when you can @strideynet